File: isAnagram/isAnagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
 var isAnagram = function(s, t) {
    if (s.length !== t.length) return false

    let letters = {}

    for (let letter of s) {
        if (letters[letter] === undefined) {
            letters[letter] = 1
        } else {
            letters[letter]++
        }
    }
    for (let letter of t) {
        if (letters[letter] === undefined) return false
        if (letters[letter] < 1) return false
        letters[letter]--
    }
    return true
};

File: isAnagram/isAnagram.py
class Solution:
    def isAnagram(self, s:str, t:str) -> bool:
        if len(s) != len(t):
            return False

        letters = {}
        for letter in s:
            if letter not in letters:
                letters[letter] = 1
            else:
                letters[letter] += 1
        for letter in t:
            if letter not in letters:
                return False
            if letters[letter] < 1:
                return False
            letters[letter] -= 1

        return True
     